### ISSUE TITLE
Pod selectors labels hardcoded in the argocd cli

### ISSUE THREAD

**Issue by Diliz** [argoproj/argo-cd#10200]
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

The argocd deployment matchLabels are hardcoded in the argocd cli:
https://github.com/argoproj/argo-cd/blob/master/cmd/argocd/commands/admin/app.go#L262
https://github.com/argoproj/argo-cd/blob/master/cmd/argocd/commands/admin/cluster.go#L82
https://github.com/argoproj/argo-cd/blob/master/cmd/argocd/commands/headless/headless.go#L54

The matchLabels in deployments can be configured in the argocd helm template:
https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-repo-server/deployment.yaml#L12
https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/_helpers.tpl#L197
https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/values.yaml#L6

So if I don't use the main names, I can't use the argocd cli for syncing or else, it wont works.

**To Reproduce**

Change the release name or one of your argocd deployment name in the argocd helm chart, then use the argocd cli to sync an app.

In values.yaml:
```
nameOverride: "toto"
```

Deploy your app, then:
```
argocd app sync my-toto-app --revision="1234567"
```

Check your matchLabels:
```
spec:
  selector:
    matchLabels:
      app.kubernetes.io/name: toto-repo-server
```

**Expected behavior**

The argocd cli should work as expected with defined values with the official argocd helm chart.

**Version**

Introduced since 2.3.x

```
argocd: v2.4.8+844f79e
  BuildDate: 2022-07-29T17:23:00Z
  GitCommit: 844f79eb9d8f3ab96d4ce6f8df211c6093a660ba
  GitTreeState: clean
  GoVersion: go1.18.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.4.8+844f79e
  BuildDate: 2022-07-29T17:23:00Z
  GitCommit: 844f79eb9d8f3ab96d4ce6f8df211c6093a660ba
  GitTreeState: clean
  GoVersion: go1.18.4
  Compiler: gc
  Platform: linux/amd64
  Kustomize Version: v4.5.5 2022-05-20T20:25:40Z
  Helm Version: v3.9.0+g7ceeda6
  Kubectl Version: v0.23.1
  Jsonnet Version: v0.18.0
```

**Logs**

The current error if the selector is mismatching :
```
FATA[0000] rpc error: code = FailedPrecondition desc = error getting repo server client: cannot find pod with selector: [app.kubernetes.io/name=argocd-repo-server]
```

**Proposal**
Add the possibility to change the pod selector matching labels with the argocd cli.

It'll fix this issue for the helm chart, as well as custom argocd reimplementations/usage (exemple: openshift-gitops from red-hat)

Thank you for reading,
Diliz

**Comment by Diliz** [comment]
Hello!

Any news about this issue? :)

**Comment by Diliz** [comment]
Hello! Is there any news regarding this issue? :)

**Comment by ivancavanaugh** [comment]
We have the same issue. Version of ArgoCD is:
argocd: v2.4.11+unknown
  BuildDate: 2022-09-07T16:16:56Z
  GitCommit:
  GitTreeState: clean
  GoVersion: go1.18.4
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.4.11+unknown

**Comment by pranayraut** [comment]
I would like to pick this up for hacktoberfest

**Comment by maheshbaliga** [comment]
Hi @pranayraut, are you working on this issue? If not, I would like to take it up.

**Comment by maheshbaliga** [comment]
Hi @crenshaw-dev, these hard-coded component names can be defined as environmental variables (name & value pair) for the respective containers; and then accessed in the CLI code. Is this approach OK?

**Comment by crenshaw-dev** [comment]
@maheshbaliga I think env vars are fine! I'd recommend also providing CLI flags which fall back to env vars. We've used that pattern for quite a few things.

**Comment by protyay** [comment]
@crenshaw-dev Hey, can you close this issue ?  It would help find new issues that need addressing. Thanks

**Comment by crenshaw-dev** [comment]
@protyay the fix hasn't been merged yet.

**Comment by Diliz** [comment]
> @crenshaw-dev Hey, can you close this issue ? It would help find new issues that need addressing. Thanks

The pr still needs a merge
 Old issues need addressing as well ;)

**Comment by maheshbaliga** [comment]
Hi @crenshaw-dev @Diliz, will try to implement the review comments at the earliest. Almost done.

**Comment by protyay** [comment]
I see. Thanks

**Comment by nmatare** [comment]
+1 for this

**Comment by Ronald-TR** [comment]
Any update about this issue? no matter how flags i use to list the apps, the `argocd` cli always returns `FATA[0002] cannot find pod with selector: [app.kubernetes.io/name=argocd-repo-server]`

**Comment by crenshaw-dev** [comment]
There's an open PR. It was originally implemented as a global CLI flag. I asked the author to re-implement as individual flags where necessary. I didn't realize how many places it would be necessary. In hindsight, I think we need to go with a global flag instead of flags on individual commands. The original PR author understandably has moved on.

If someone wants to pick up the PR, there old "global flag" state might still be in that fork/branch history. It would just need to be rebased.

**Comment by maheshbaliga** [comment]
Hi @crenshaw-dev. Sorry, I'm unable to allocate time to the PR because I am currently occupied with other tasks and do not have the previous version of the code too. I will need time to address the PR again, but I'm open to someone else taking up the issue in the meantime.

**Comment by omegion** [comment]
We are also facing the same problem as we manage x2 ArgoCD instances in one cluster.

**Comment by crenshaw-dev** [comment]
Ah! The fix is implemented. It will be available in 2.9. https://github.com/argoproj/argo-cd/pull/14605

**Comment by mtougeron** [comment]
@crenshaw-dev any chance that this will be back-ported to the 2.8 series?
